### PR TITLE
Feature/email serivce

### PR DESCRIPTION
**Pull request by agustinpurita**
## Pull request type

- [x] Feature

### CODE REVIEW

**Review by Alxzu — COMMENTED** [review]
This looks awesome! 👏 . Can you also change the README to reflect some of the new folders that you add such as "interfaces"? Also, take a look at spaces between interfaces definitions

File: src/utils/email/emailPlatformMap.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { EmailPlatformConstants } from '@constants/email';
+import { EmailService } from '@services/email.service';
+import { Transporter, SentMessageInfo } from 'nodemailer';
+
+export const transpoterMapper: {
+  [k in EmailPlatformConstants.EmailPlatforms]: () => Transporter<SentMessageInfo>;
```
Comment (alexfradiani) [inline src/utils/email/emailPlatformMap.ts]: this signature definition looks a bit complex. Can it be encapsulated into its own type?

**Review by alexfradiani — CHANGES_REQUESTED** [review]

File: src/utils/email/emailPlatformMap.ts (hunk 1 above)
Comment (alexfradiani) [reply src/utils/email/emailPlatformMap.ts]: also, there is a typo in "transpoter"

**Comment by agustinpurita** [comment]
> This looks awesome! 👏 . Can you also change the README to reflect some of the new folders that you add such as "interfaces"? Also, take a look at spaces between interfaces definitions

Yes! Good points.
Changes made, thanks!

File: src/utils/email/emailPlatformMap.ts (hunk 1 above)
Comment (agustinpurita) [reply src/utils/email/emailPlatformMap.ts]: Changed!

**Review by Alxzu — APPROVED** [review]

**Review by alexfradiani — APPROVED** [review]

File: src/interfaces/email/email.interface.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export interface IEmail {
+  from: string;
+  to: string;
+  subject: string;
+  template?: string;
+  text?: string;
+  context?: any;
+}
+
+export interface AuthSendGrid {
```
Comment (zk182) [inline src/interfaces/email/email.interface.ts]: probably`IAuthSendGrid`. I also used the `I` convention to name interfaces, but recently I was reading clean code and they had a different point of view, if you want to check it out :

http://www.amygdalum.net/en/naming-interface-implementation.html

https://softwareengineering.stackexchange.com/questions/117348/should-interface-names-begin-with-an-i-prefix

**Review by zk182 — APPROVED** [review]
good work!

File: src/interfaces/email/email.interface.ts (hunk 2 above)
Comment (agustinpurita) [reply src/interfaces/email/email.interface.ts]: oh! very useful information! I'll take a look at it, for now I changed it to `IAuthSendGrid`.
thanks for your review!

**Pull request merged**